### PR TITLE
[MakeRegistration] add support for verify email attributes

### PR DESCRIPTION
**Pull request by jrushlow**
- fixes CI related MakeRegistrationFailures
- Adds support for class level attributes, specifically `UniqueEntity` attributes on the `User` entity.
- Adds support for VerifyEmail property attributes

Known Issues:
Adding a class level attribute that utilizes an aliased use statement, e.g. `use CLASS as PREFIX` type alias is not supported in this PR.

### CODE REVIEW

**Comment by jrushlow** [comment]
Tests are needed for CSM

File: src/Util/ClassSourceManipulator.php [hunk 1]
```diff
@@ -900,6 +925,31 @@ public function addUseStatementIfNecessary(string $class): string
...
 
+    /**
+     * Builds a PHPParser attribute node.
+     *
+     * @param ClassNameDetails $attributeClass The attribute class which should be used for the attribute. If the class has a substitute
+     *                                         prefix, the node is built using the prefix. E.g. #[ORM\Column()] Otherwise, the
+     *                                         short name is used. E.g. #[Column()]
+     * @param array            $options        the named arguments for the attribute ($key = argument name, $value = argument value)
+     */
+    public function buildAttributeNode(ClassNameDetails $attributeClass, array $options): Node\Attribute
```
Comment (weaverryan) [inline src/Util/ClassSourceManipulator.php:936]: If I understand correctly, we're using `ClassNameDetails` here so that we can communicate a possible prefix - e.g. `ORM\`. If so, could we instead add that as a 3rd arg?

```suggestion
    public function buildAttributeNode(string $attributeClass, array $options, string $attributePrefix = null): Node\Attribute
```

**Review by weaverryan — COMMENTED** [review]
This generally looks good to me 👍 

The ClassSourceManipulator is a mess when it comes to attributes vs annotations, but that's a known issue that we need to fix... at some point.

File: src/Util/ClassSourceManipulator.php (hunk 1 above)
Comment (jrushlow) [reply src/Util/ClassSourceManipulator.php:936]: Yes, but also because we need access to `ClassNameDetails::getShortName()`. Otherwise we get a FQCN in the in the attribute.. e.g. `#[Doctrine\Some\Long\Path\MyAttribute(....)]`. 

I suppose we could do something like:
```php
public function buildAttributeNode(string $attributeClass, array $options, string $attributePrefix = null): Node\Attribute
{
    $details = new ClassNameDetails($attributeClass, '\\');
}
```

And I think I was first going with that, except that the attribute method is then making assumptions about the 2nd argument for the CSD constructor - Minor detail which may be irrelevant in the grand scheme of things..

Comment (jrushlow) [reply src/Util/ClassSourceManipulator.php:936]: Wait we also have MakerBundles `Str::getShortClassName()` - https://github.com/symfony/maker-bundle/blob/64318ec4a7870904e54f70e76e29ac8bc20e9875/src/Str.php#L123

That would work even better then init'ing a ClassNameDetails object..

File: src/Util/ClassSourceManipulator.php [hunk 2]
```diff
@@ -358,6 +367,17 @@ public function addProperty(string $name, array $annotationLines = [], $defaultV
         $this->addNodeAfterProperties($newPropertyNode);
     }
 
+    public function addAttributeToClass(string $attributeClass, array $options): void
```
Comment (jrushlow) [inline src/Util/ClassSourceManipulator.php:370]: Adding a class level attribute that requires an aliased use statement, e.g. ` use Doctrine\Mapping as ORM` is not possible in this scope without refactoring the `ClassSourceManipulator::addUseStatementIfNecessary()` method. A task which is not required by our current use case and should be handled in a separate PR.
